### PR TITLE
More formatting changes, using constructors now

### PR DESCRIPTION
**Pull request by amartinez-cg**
This implements constructors wherever possible for all function headers.

### CODE REVIEW

File: src/shaders/MaterialX/color2.h [hunk 1]
```diff
@@ -15,340 +15,287 @@ struct color2
 
 color2 make_color2(float r, float g)
```
Comment (lgritz) [inline src/shaders/MaterialX/color2.h:16]: Is make_color2 even needed now that we have constructors? Is there any use of make_color(r,g) that should not simply be color2(r,g)?

File: src/shaders/MaterialX/color2.h [hunk 2]
```diff
@@ -15,340 +15,287 @@ struct color2
...
-    color2 out;
-    out.r = fg.r * (1 - bg.g);
-    out.g = fg.g * (1 - bg.g);
-    return out;
+    return color2 (fg.r * (1 - bg.g),
+                   fg.g * (1 - bg.g)
+                   );
 }
 
 color2 over(color2 fg, color2 bg)
```
Comment (lgritz) [inline src/shaders/MaterialX/color2.h:273]: Is this used? It sure is weird. For a color2 (where you have called the components r and g) does it make sense to magically treat g as if it were alpha?

File: src/shaders/MaterialX/color4.h [hunk 3]
```diff
@@ -15,312 +15,267 @@ struct color4
...
+    return invert(in, 1.0);
 }
 
 color4 clamp(color4 in, color4 low, color4 high)
 {
-    color4 out = in;
-    out.rgb = clamp(in.rgb, low.rgb, high.rgb);
-    out.a = clamp(in.a, low.a, high.a);
-    return out;
+    return color4 (clamp(in.rgb, low.rgb, high.rgb),
```
Comment (lgritz) [inline src/shaders/MaterialX/color4.h:156]: Silly question (maybe for Doug): for an RGB+alpha, is the right definition of "invert" to also invert the alpha? When would that be helpful? Or should alpha be left alone and only the color inverted? Er, or is the right thing that invert(rgba) == (a-rgba, a), in other words "invert" only with respect to the total coverage?The same question might be asked of other color4 functions.

The unobviousness of exactly what the math should mean on a color+alpha is a big part of why I didn't make a color4 as a fundamental type in the language. :-)

**Comment by amartinez-cg** [comment]
There actually *isn't* any use of make_color2, or any of the make_<TYPE>.
They can all be removed.

File: src/shaders/MaterialX/color4.h (hunk 3 above)
Comment (lgritz) [reply src/shaders/MaterialX/color4.h:156]: Let me phrase it an alternative way.

For a pixel that's 100% covered (i.e. viewing an opaque object) and white, is the "inverse" a pixel that's covered and black? Or a pixel that's uncovered (completely transparent)?

For a pixel that's empty (rgba==0), what is the inverse? Is it a pixel that's opaque and white? A pixel that's still uncovered, but glowing bright white? Or does invert not change a pixel that "has nothing in it"?

For a pixel that's 3/4 covered with a white thing... so its rgba = (0.75,0.75,0.75,0.75): is its inverse a pixel that/s 3/4 covered with a black thing (0,0,0,.75)?  Or is it a pixel that's (0.25,0.25,0.25,0.25), which is a pixel that's only 1/4 covered, but still by something completely white? That seems wrong? Or something else?

We can certainly table the discussion about what math means for color4 for another PR and another topic. It shouldn't stop you from committing the constructor PR.

**Comment by amartinez-cg** [comment]
My sense is that MaterialX is treating the color2 type as if it were "A
grey-scale image, with an alpha channel."  If that is the case, then maybe
color2 should be defined as { float r; float a;}  ?

**Comment by lgritz** [comment]
The use of constructors LGTM. Squash and merge at will.
But let's circle back with Doug about the color4 math. It's not just invert that's a problem.

**Comment by lgritz** [comment]
Yes, I agree that if color2 means gray + alpha, then for the sake of clarity I would recommend naming the members r and a.  Ick. Maybe gray + a? Unless that will make confusion because on GPUs, the swizzling is already called .r and .g. Sigh.

**Comment by amartinez-cg** [comment]
I totally agree.  We had channel masks once.... just saying.

Lets bring this to the mx guys.

**Pull request merged**